src/app/externs/backendapi.js

```diff
@@ -94,6 +94,13 @@ backendApi.ReplicaSetPod;
 
 /**
  * @typedef {{
+ *   name: string
+ * }}
+ */
+backendApi.NamespaceSpec;
```

Comment:
You've defined a nice pattern here. The `Spec` suffix. I like it for POST REST handlers. Can you also rename `AppDeployment` to `AppDeploymentSpec`? This will make the code consistent.